platform/core.output2/src/org/netbeans/core/output2/NbIOProvider.java

```diff
@@ -214,7 +214,17 @@ public Lookup getIOLookup(InputOutput io) {
     @Override
     public void resetIO(InputOutput io) {
         if (io instanceof NbIO) {
-            ((NbIO) io).reset();
+            if(io.getOut() instanceof NbWriter) {
+                try {
+                    // NbWriter does OutWriter fixup,
+                    // then NbWriter invokes ((NbIO) io).reset()
+                    io.getOut().reset();
```

Comment:
The comment seems correct. At first I thought maybe io.getIn().reset() would also need to be called, but as the comment says, we the implementation of NbWriter will call NbIO.reset() in turn..